File: src/packets/PacketS2CMediaMetadata.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x36
 * Media metadata
 * Sends info about the current playing track to clients
 * Data:
 *      (Unsigned Char) SourceID
 *      (Short) Time
 *      (String) Text
 */
class PacketS2CMediaMetadata extends PacketS2C {
    constructor(sourceID, title, artist, artworkURL) {
        super();

        this._buffer.writeUInt8(sourceID);
        this._buffer.writeStringNT(title ? title : "");
        this._buffer.writeStringNT(artist ? artist : "");
        this._buffer.writeStringNT(artworkURL ? artworkURL : "");
    }

    getID() {
        return 0x36;
    }
}

module.exports = PacketS2CMediaMetadata;

File: src/RNetPi.Core/Packets/PacketS2CMediaMetadata.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x36
/// Media Metadata
/// Sends info about the current playing track to clients
/// Data:
///     (Unsigned Char) SourceID
///     (String) Title
///     (String) Artist
///     (String) Artwork URL
/// </summary>
public class PacketS2CMediaMetadata : PacketS2C
{
    public const byte ID = 0x36;

    public PacketS2CMediaMetadata(byte sourceID, string? title, string? artist, string? artworkURL)
    {
        Writer.Write(sourceID);
        WriteNullTerminatedString(title ?? string.Empty);
        WriteNullTerminatedString(artist ?? string.Empty);
        WriteNullTerminatedString(artworkURL ?? string.Empty);
    }

    public override byte GetID() => ID;
}